File: src/julia/modules/Multistart/src/improvement.jl
function updateCurrentResults!(extmd::ExternalMSLPData, _::AllParams)::Nothing
	# extmd.timeToBest = (CPUtime_us() - extmd.startTime) / 1e6
	extmd.timeToBest = cpu_times()[1] - extmd.startTime
	extmd.iterationToBest = extmd.iteration
	extmd.bestSol = extmd.currSol
	update_offset = extmd.iteration - extmd.iterationToBest
	extmd.largestUpdateOffset = max(extmd.largestUpdateOffset, update_offset)
	extmd.countImprovements += 1

	@printf(
		"%.2f;%.2f;%d;%.6f\n",
		extmd.bestSol.value,
		extmd.lastSGreedySolValue,
		extmd.iteration,
		extmd.timeToBest
	)
    return nothing
end

File: src/python/modules/multistart/improvement.py
import time

from .entities_mslp import ExternalMSLPData

def update_current_results(extmd: ExternalMSLPData, _) -> None:
    extmd.time_to_best = time.time() - extmd.start_time
    extmd.iteration_to_best = extmd.iteration
    extmd.best_sol = extmd.curr_sol
    update_offset = extmd.iteration - extmd.iteration_to_best
    extmd.largest_update_offset = max(extmd.largest_update_offset, update_offset)

    print(
        f"{extmd.best_sol.value:.2f};"
        f"{extmd.curr_sol.value:.2f};"
        f"{extmd.iteration};"
        f"{extmd.time_to_best:.6f}"
    )
